typescript/lib/config.ts

```diff
@@ -288,6 +288,8 @@ export class AIConfigRuntime implements AIConfig {
    * that can be used to call the GPT-4 API. It will combine all the parameters, references and metadata specified in the AIConfig."
    */
   public async resolve(promptName: string, params: JSONObject = {}) {
+    const startEvent = new CallbackEvent("on_resolve_start", __filename, {promptName, params});
```

Comment:
Something to look into as a potential followup: instead of requiring fileName passed through, we could probably do a bit of a hack in the `runCallbacks` implementation to grab the exact place it was called using a temp error stack -- https://stackoverflow.com/questions/28631260/how-to-get-filename-and-line-number-of-where-a-function-is-called-in-node

Reply:
Could also do `className: this.name or this.constructor.name` instead of __fileName here. Looking over next PR, it seems mainly beneficial for distinguishing between parser classes

Reply:
that's pretty cool, will add to linear todo as well